### DISCUSSION TITLE
Silent coverage output in terminal

### DISCUSSION THREAD

**Discussion by otakustay** [vitest-dev/vitest#4013] [Q&A]
Hi, I want vitest to output test results in json, so I use `--reporter=json`, however with `--coverage.enabled --coverage.clean --coverage.reporter=json` flags, vitest adds a line " % Coverage report from v8" after json output.

I'd like to see coverage written to `coverage/coverage-final.json` without any output in terminal, is there any flags I'm missing?

**Comment by AriPerkkio** [comment]
Currently that message is logged always when coverage is enabled. It doesn't really make sense to log it if `text` or `text-summary` reporters are not used.

**Reply by otakustay** [reply]
This prevents us to simple `JSON.parse(stdout)`, so I'd prefer not logging this line when no text report is enabled for coverage

**Reply by AriPerkkio** [reply]
> This prevents us to simple `JSON.parse(stdout)`

Why would you try to parse `stdout` as JSON? It contains all kinds of text and not just the coverage logging.

Use `--coverage.reporter=json` and read the generated `coverage/coverage-final.json` instead.

**Reply by AriPerkkio** [reply]
There's now a PR for this: https://github.com/vitest-dev/vitest/pull/4027

**Comment by otakustay** [comment]
I need the test result (from `--report=json`) as well as a coverage report (from `--coverage.reporter=json`).

The test result looks like this and seems I can only get it from terminal stdout:

```json
{
  "numTotalTestSuites": 2,
  "numPassedTestSuites": 2,
  "numFailedTestSuites": 0,
  "numPendingTestSuites": 0,
  "numTotalTests": 5,
  "numPassedTests": 5,
  "numFailedTests": 0,
  "numPendingTests": 0,
  "numTodoTests": 0,
  "startTime": 1693041503488,
  "success": true,
  "testResults": [ /* ... */ ]
}
```

**Reply by AriPerkkio** [reply]
> The test result looks like this and seems I can only get it from terminal stdout:

`--outputFile=test-results.json` should do that. 

https://vitest.dev/config/#outputfile

**Reply by otakustay** [reply]
Oh thanks, I can read json from file now
